pkg/instrumentation/nethttp/client/client_hook.go

```diff
@@ -0,0 +1,153 @@
+	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/pkg/instrumentation/nethttp/semconv"
+	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/pkg/instrumentation/shared"
+)
+
+const (
+	otelExporterPrefix     = "OTel OTLP Exporter Go"
+	instrumentationName    = "github.com/open-telemetry/opentelemetry-go-compile-instrumentation/pkg/instrumentation/nethttp"
+	instrumentationVersion = "0.1.0"
+)
+
+var (
+	logger     = shared.GetLogger()
+	tracer     trace.Tracer
+	propagator propagation.TextMapPropagator
+	initOnce   sync.Once
+)
+
+func initInstrumentation() {
+	initOnce.Do(func() {
+		if err := shared.SetupOTelSDK(); err != nil {
+			logger.Error("failed to setup OTel SDK", "error", err)
+		}
+		tracer = otel.GetTracerProvider().Tracer(
+			instrumentationName,
+			trace.WithInstrumentationVersion(instrumentationVersion),
+		)
+		propagator = otel.GetTextMapPropagator()
+		logger.Info("HTTP client instrumentation initialized")
+	})
+}
+
+// netHttpClientEnabler controls whether client instrumentation is enabled
+type netHttpClientEnabler struct{}
+
+func (n netHttpClientEnabler) Enable() bool {
+	return shared.Instrumented("NETHTTP")
+}
+
+var clientEnabler = netHttpClientEnabler{}
+
+func BeforeRoundTrip(ictx inst.HookContext, transport *http.Transport, req *http.Request) {
+	if !clientEnabler.Enable() {
+		logger.Debug("HTTP client instrumentation disabled")
+		return
+	}
+
+	// Filter out OTel exporter requests to prevent infinite loops
+	if strings.HasPrefix(req.Header.Get("User-Agent"), otelExporterPrefix) {
+		logger.Debug("Skipping OTel exporter request", "user_agent", req.Header.Get("User-Agent"))
+		return
+	}
+
+	initInstrumentation()
+
+	logger.Debug("BeforeRoundTrip called",
+		"method", req.Method,
+		"url", req.URL.String(),
+		"host", req.Host)
+
+	ctx := req.Context()
+
+	// Get trace attributes from semconv
+	attrs := semconv.HTTPClientRequestTraceAttrs(req)
+
+	// Start span
+	spanName := req.Method
+	ctx, span := tracer.Start(ctx,
+		spanName,
+		trace.WithSpanKind(trace.SpanKindClient),
+		trace.WithAttributes(attrs...),
+	)
+
+	// Inject trace context into request headers
+	propagator.Inject(ctx, propagation.HeaderCarrier(req.Header))
+
+	// Update request with new context
+	newReq := req.WithContext(ctx)
+	ictx.SetParam(1, newReq)
+
+	// Store data for after hook
+	ictx.SetData(map[string]interface{}{
+		"ctx":   ctx,
+		"span":  span,
+		"req":   req,
+		"start": time.Now(),
+	})
+}
+
+func AfterRoundTrip(ictx inst.HookContext, res *http.Response, err error) {
+	if !clientEnabler.Enable() {
+		return
+	}
+
+	data, ok := ictx.GetData().(map[string]interface{})
+	if !ok || data == nil {
+		logger.Debug("AfterRoundTrip: no data from before hook")
+		return
+	}
+
+	span, ok := data["span"].(trace.Span)
+	if !ok || span == nil {
+		logger.Debug("AfterRoundTrip: no span from before hook")
+		return
+	}
+	defer span.End()
+
+	startTime, _ := data["start"].(time.Time)
```

Comment:
This is only used in `if res != nil {` branch, so it would be better to move it there
```suggestion
if res != nil {
	startTime, _ := data["start"].(time.Time)
```